src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -7564,5 +7564,42 @@ public void BrokenSwitchExpression_03()
             }
             EOF();
         }
+
+        [Fact, WorkItem(33675, "https://github.com/dotnet/roslyn/issues/33675")]
+        public void ParenthesizedNamedConstantPatternInSwitchExpression()
+        {
+            UsingExpression("e switch { (X) => 1 }");
```

Comment:
>UsingExpression("e switch { (X) => 1 }"); [](start = 12, length = 41)

Please add a unit test that executes the scenario from the issue and verifies expected runtime behavior.

Reply:
I think we already have tests for binding and code gen of parenthesized constants, but I'll add another if you think it is valuable.